components/CreateResourceModal/CreateResourceModal.jsx

```diff
@@ -101,15 +105,47 @@ export default class CreateResourceModal extends React.Component {
     });
   }
 
+  togglePoll = async () => {
+    this.setState({
+      isPoll: !this.state.isPoll,
+    });
+  }
+
+  updatePoll = async (data) => {
+    this.setState({
+      isPoll: !!data,
+      pollData: data,
+    });
+  }
+
   render() {
     var {
       onClose,
       isModalOpen,
-      submitButtonText
+      submitButtonText,
+      loggedInUser,
+      existingPoll
     } = this.props;
 
     if (!submitButtonText) submitButtonText = 'Submit';
 
+    if (existingPoll && !this.state.pollData) {
+      this.state.pollData = this.state.pollData || existingPoll;
+      this.state.isPoll = !!this.state.pollData;
+    }
+
+    const { height, width } = Dimensions.get('window');
+    let modalHeight;
+    if (this.state.isPoll) {
```

Comment:
Do you mean the top of the modal going off screen? If so, that happens in Android too, it's just because the modal is too tall. I think it's OK because all the text inputs are still on screen / can be scrolled to show on screen, and I don't want to make the modal shorter because I'd like to keep enough space to show question and options.
On Facebook the top of the modal stays where it is and the modal becomes shorter, and the poll is scrolled to the cursor. I tried to implement this behaviour but could not figure out how to do it.

Reply:
My PR should fix this anyways, so it's not a big deal. I'm working on updating the create post UI